src/bit_str.jl

```diff
@@ -1,124 +1,98 @@
-export BitStr, @bit_str, bcat, bit_literal, bit, to_location, onehot, onehot_batch
+export BitStr, @bit_str, bcat, bit_literal, to_location, onehot, onehot_batch
 
 """
-    BitStr{T}
+    BitStr{N} <: Integer
 
-String literal for bits.
+primitive type for bit string with fixed length `N`, implements an integer with 64 bit storage.
 
-    BitStr(value[, len=ndigits(value)])
+    BitStr{N}(value)
 
-Returns a `BitStr`, by default the length is set to the minimum length required to represent
-`value` as bits.
-
-    BitStr(str::String)
-
-Parse the input string to a BitStr. See [`@bit_str`](@ref) for more details.
+Returns a `BitStr`.
 
 ## Example
 
 `BitStr` supports some basic arithmetic operations. It acts like an integer, but supports
 some frequently used methods for binary basis.
 
 ```julia
-julia> bit"101" * 2
-1010 (10)
```

Comment:
Then it will be ugly in measure results, which is an array. e.g.
```
bit"1000011"
bit"1000011"
bit"1000011"
bit"1000011"
bit"1000011"
bit"1000011"
bit"1000011"
```

I wish it can be simpler, without `""`

Reply:
How do you like `0111₂`, means bit string is a basis 2 representation of integer.

Reply:
I think most places will have a parathesis for bits, e.g 0101_{(2)} , we can print this as

```jl
101010₍₂₎
```

what do you think?

Reply:
I think it is ok, but is it possible to input it in Julia?

Reply:
Yes, just try it.

Reply:
Yes, just try it.

Reply:
Ok, I see.